src/main/java/com/example/gradecalculator/web/controller/MyErrorController.java

```diff
@@ -0,0 +1,36 @@
+package com.example.gradecalculator.web.controller;
+
+import jakarta.servlet.RequestDispatcher;
+import jakarta.servlet.http.HttpServletRequest;
+import org.springframework.boot.web.servlet.error.ErrorController;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+@Controller
+public class MyErrorController implements ErrorController {
+
+    @RequestMapping("/error")
+    public String handleError(HttpServletRequest request, Model model) {
+        Object status = request.getAttribute(RequestDispatcher.ERROR_STATUS_CODE);
+
+        if (status != null) {
+            Integer statusCode = Integer.valueOf(status.toString());
+
+            if(statusCode == HttpStatus.NOT_FOUND.value()) {
+                model.addAttribute("message", "Error " + statusCode + ": Page not found");
+            }
+            else if(statusCode == HttpStatus.INTERNAL_SERVER_ERROR.value()) {
+                model.addAttribute("message", "Error " + statusCode + ": Internal server error");
+            }
+            else {
+                model.addAttribute("message", "Error " + statusCode + ": An error occurred");
+            }
+        }
+        else {
+            model.addAttribute("message", "An unexpected error occurred");
+        }
```

Comment:
please refactor this into a switch-case for better readabilty. also try to reduce duplicated code

Reply:
done

Reply:
better, but still not happy about error message. This can be improved
It should look like this at the end:
`model.addAttribute("message", errormessage);`

Also: do not override the status Object

Reply:
done and I hope the code matches your expections

Reply:
Now its back to the original state ;D
Here is how I would do it:

```
    @RequestMapping("/error")
    public String handleError(HttpServletRequest request, Model model) {
        Object status = request.getAttribute(RequestDispatcher.ERROR_STATUS_CODE);
        String errorMessage = getErrorMessage(status);
        model.addAttribute("message", errorMessage);
        return "error";
    }

    private static String getErrorMessage(Object status) {
        String errorMessage;
        if (status != null) {
            Integer statusCode = (Integer) status;
            String statusMessage = switch (statusCode) {
                case 404 -> "Page not found";
                case 500 -> "Internal server error";
                default -> "An error occurred";
            };
            errorMessage = "Error " + statusCode + ": " + statusMessage;
        } else {
            errorMessage = "An unexpected error occurred";
        }
        return errorMessage;
    }
```

Reply:
Which can again be refactored into:

```
    private static String getErrorMessage(Object status) {
        if (status == null) {
            return "An unexpected error occurred";
        }
        Integer statusCode = (Integer) status;
        String statusMessage = switch (statusCode) {
            case 404 -> "Page not found";
            case 500 -> "Internal server error";
            default -> "An error occurred";
        };
        return "Error " + statusCode + ": " + statusMessage;
    }
``` 

🤯

Reply:
Okay if have done it like you did, but I would never come to this